packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.test.ts

```diff
@@ -1,79 +1,48 @@
 import { describe, it } from 'node:test';
 import assert from 'assert';
 import { PackageManagerControllerFactory } from './package_manager_controller_factory.js';
-import { type PackageManagerController } from './package_manager_controller.js';
-
-type PackageManagerControllerType = PackageManagerController & {
-  executable: string;
-  binaryRunner: string;
-  addTypescript?: boolean;
-  addLockFile?: boolean;
-  installCommand?: string;
-};
+import { PackageManagerController } from './package_manager_controller.js';
+import { NpmPackageManagerController } from './npm_package_manager_controller.js';
 
 void describe('packageManagerControllerFactory', () => {
   const packageRoot = '/path/to/project';
 
   void describe('getPackageManagerController', () => {
-    void it('should return the correct package manager controller for npm', () => {
-      const packageManagerControllerFactory =
-        new PackageManagerControllerFactory(
-          packageRoot,
-          'npm/7.0.0 node/v15.0.0 darwin x64'
-        );
-
-      const packageManagerController =
-        packageManagerControllerFactory.getPackageManagerController() as PackageManagerControllerType;
-      assert.match(packageManagerController.executable, /npm/);
-      assert.match(packageManagerController.binaryRunner, /npx/);
-      assert.equal(packageManagerController.addTypescript, undefined);
-      assert.equal(packageManagerController.addLockFile, undefined);
-    });
-
-    void it('should return the correct package manager controller for pnpm', () => {
-      const packageManagerControllerFactory =
-        new PackageManagerControllerFactory(
-          packageRoot,
-          'pnpm/5.0.0 node/v15.0.0 darwin x64'
+    const testCases = [
+      {
+        name: 'npm',
+        userAgent: 'npm/7.0.0 node/v15.0.0 darwin x64',
+        expectedInstanceof: NpmPackageManagerController,
+      },
+      {
+        name: 'pnpm',
+        userAgent: 'pnpm/5.0.0 node/v15.0.0 darwin x64',
+        expectedInstanceof: PackageManagerController,
```

Comment:
```suggestion
        expectedInstanceof: PnpmPackageManagerController,
```
Same applies to the yarn classic and yarn modern test case as well

Reply:
Good catch!